socorro/unittest/external/sqs/test_crashqueue.py

```diff
@@ -0,0 +1,94 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+import pytest
+
+from socorro.external.sqs.crashqueue import SQSCrashQueue
+from socorro.lib.ooid import create_new_ooid
+from socorro.unittest.external.sqs import get_sqs_config
+
+
+VISIBILITY_DEADLINE = 2
+
+
+class TestSQSCrashQueue:
+    def test_iter(self, sqs_helper):
+        standard_crash = create_new_ooid()
+        sqs_helper.publish("standard", standard_crash)
+
+        reprocessing_crash = create_new_ooid()
+        sqs_helper.publish("reprocessing", reprocessing_crash)
+
+        priority_crash = create_new_ooid()
+        sqs_helper.publish("priority", priority_crash)
+
+        crash_queue = SQSCrashQueue(get_sqs_config())
+        new_crashes = list(crash_queue.new_crashes())
+
+        # Assert the shape of items in new_crashes
+        for item in new_crashes:
+            assert isinstance(item, tuple)
+            assert isinstance(item[0], tuple)  # *args
+            assert isinstance(item[1], dict)  # **kwargs
+            assert list(item[1].keys()) == ["finished_func"]
+
+        # Assert new_crashes order is the correct order
+        crash_ids = [item[0][0] for item in new_crashes]
+        assert crash_ids == [priority_crash, standard_crash, reprocessing_crash]
+
+    def test_ack(self, sqs_helper):
+        original_crash_id = create_new_ooid()
+
+        # Publish crash id to the queue
+        sqs_helper.publish("standard", original_crash_id)
+
+        crash_queue = SQSCrashQueue(get_sqs_config())
+        new_crashes = list(crash_queue.new_crashes())
+
+        # Assert original_crash_id is in new_crashes
+        crash_ids = [item[0][0] for item in new_crashes]
+        assert crash_ids == [original_crash_id]
+
+        # Now call it again; note that we haven't acked the crash_ids
+        # nor have the leases expired
+        second_new_crashes = list(crash_queue.new_crashes())
+        assert second_new_crashes == []
+
+        # Now ack the crash_id and we don't get it again
+        for args, kwargs in new_crashes:
+            kwargs["finished_func"]()
+
+        # Wait beyond the ack deadline in the grossest way possible
+        time.sleep(VISIBILITY_DEADLINE + 1)
```

Comment:
This isn't right. This is a visibility deadline that's not related to the default visibility for the queue.